integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,622 @@
+  var DC_FLOODLIGHT_ID = this.options.DC_FLOODLIGHT_ID;
+  var accountId =
+    GA_WEB_MEASUREMENT_ID ||
+    GA_WEB_APP_MEASUREMENT_ID ||
+    AW_CONVERSION_ID ||
+    DC_FLOODLIGHT_ID;
+
+  if (GA_WEB_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (GA_WEB_APP_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_APP_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_APP_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (AW_CONVERSION_ID) {
+    config.push(['config', AW_CONVERSION_ID]);
+  }
+
+  if (DC_FLOODLIGHT_ID) {
+    config.push(['config', DC_FLOODLIGHT_ID]);
+  }
+
+  if (accountId) {
+    this.load({ accountId: accountId }, function() {
+      // Default routing.
+      for (var i = 0; i < config.length; i++) {
+        push(config[i][0], config[i][1]);
+      }
+      that.ready();
+    });
+  } else {
+    // Error case where not any of the ID specified
+  }
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+
+GTAG.prototype.loaded = function() {
+  return !!(
+    window.gtagDataLayer && Array.prototype.push !== window.gtagDataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} event
+ */
+
+GTAG.prototype.identify = function(identify) {
+  var userId = identify.userId();
+  if (userId) {
+    if (this.options.GA_WEB_MEASUREMENT_ID) {
+      push('config', this.options.GA_WEB_MEASUREMENT_ID, {
+        user_id: userId
+      });
+    }
+    if (this.options.GA_WEB_APP_MEASUREMENT_ID) {
+      push('config', this.options.GA_WEB_APP_MEASUREMENT_ID, {
+        user_id: userId
+      });
+    }
+  }
+};
+
+/**
+ * Track
+ *
+ * @api public
+ * @param {Track} track
+ */
+
+GTAG.prototype.track = function(track) {
+  var options = this.options;
+  var props = track.properties();
+  props.event = track.event() || '';
+
+  var gaOptions = this.options.gaOptions || {};
+  if (gaOptions && Object.keys(gaOptions).length) {
+    if (gaOptions.setAllMappedProps) {
+      // set custom dimension and metrics if present
+      // REF: https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets
+
+      var customMap = merge(gaOptions.dimensions, gaOptions.metrics);
+      if (options.GA_WEB_MEASUREMENT_ID) {
+        push('config', this.options.GA_WEB_MEASUREMENT_ID, {
+          custom_map: customMap
+        });
+      }
+      if (options.GA_WEB_APP_MEASUREMENT_ID) {
+        push('config', this.options.GA_WEB_APP_MEASUREMENT_ID, {
+          custom_map: customMap
+        });
+      }
```

Comment:
@gpsamson Both are referrenced from existing google analytics integration, for initialise set custom dimension while initialising the integration which will be referred for all subsequent event calls(I think instead of **config** i need to use **set** that will persist further), and for track method user can override the existing set metrics if wish to.